File: test/data/AttendanceDAOTest.py
from data.AttendanceDAO import AttendanceDAO

print("Running AttendanceDAO Test...")

dao = AttendanceDAO()
dao.save_attendance("STU301", "2026-01-24", "Absent")
dao.get_attendance_by_student("STU301")

print("AttendanceDAO Test Completed.")

File: test/data/AttendanceDAOTest.java
package test.data;

import data.AttendanceDAO;

/**
 * Test class for AttendanceDAO
 */
public class AttendanceDAOTest {

    public static void main(String[] args) {

        // Arrange
        AttendanceDAO dao = new AttendanceDAO();

        // Act
        System.out.println("Running AttendanceDAO Test...");
        dao.saveAttendance("STU301", "2026-01-24", "Absent");
        dao.getAttendanceByStudent("STU301");

        // Assert (manual verification via console output)
        System.out.println("AttendanceDAO Test Completed.");
    }
}

